tia-client/src/main/java/com/teamscale/tia.client/CommandLineInterface.java

```diff
@@ -0,0 +1,153 @@
+
+	private final List<String> arguments;
+	private final String command;
+	private final ITestwiseCoverageAgentApi api;
+
+	public CommandLineInterface(String[] arguments) {
+		this.arguments = new ArrayList<>(Arrays.asList(arguments));
+		if (arguments.length < 2) {
+			throw new InvalidCommandLineException(
+					"You must provide at least two arguments: the agent's URL and the command to execute");
+		}
+
+		HttpUrl url = HttpUrl.parse(this.arguments.remove(0));
+		api = ITestwiseCoverageAgentApi.createService(url);
+
+		command = this.arguments.remove(0);
+	}
+
+	/** Entry point. */
+	public static void main(String[] arguments) throws Exception {
+		new CommandLineInterface(arguments).runCommand();
+	}
+
+	private void runCommand() throws Exception {
+		switch (command) {
+			case "startTestRun":
+				startTestRun();
+				break;
+			case "startTest":
+				startTest();
+				break;
+			case "endTest":
+				endTest();
+				break;
+			case "endTestRun":
+				endTestRun();
+				break;
+			default:
+				throw new InvalidCommandLineException(
+						"Unknown command '" + command + "'. Should be one of startTestRun, startTest, endTest," +
+								" endTestRun");
+		}
+	}
+
+	private void endTestRun() throws Exception {
+		AgentCommunicationUtils.handleRequestError(api.testRunFinished(),
+				"Failed to create a coverage report and upload it to Teamscale. The coverage is most likely lost");
+	}
+
+	private void endTest() throws Exception {
+		if (arguments.size() < 2) {
+			throw new InvalidCommandLineException(
+					"You must provide the uniform path of the test that is about to be started" +
+							" as the first argument of the endTest command and the test result as the second.");
+		}
+		String uniformPath = arguments.remove(0);
+		ETestExecutionResult result = ETestExecutionResult.valueOf(arguments.remove(0).toUpperCase());
+
+		String message = readStdin();
+
+		// the agent already records test duration, so we can simply provide a dummy value here
+		TestExecution execution = new TestExecution(uniformPath, 0L, result, message);
+		AgentCommunicationUtils.handleRequestError(api.testFinished(uniformPath, execution),
+				"Failed to end coverage recording for test case " + uniformPath +
+						". Coverage for that test case is most likely lost.");
+	}
+
+	private void startTest() throws Exception {
+		if (arguments.size() < 1) {
+			throw new InvalidCommandLineException(
+					"You must provide the uniform path of the test that is about to be started" +
+							" as the first argument of the startTest command");
+		}
+		String uniformPath = arguments.remove(0);
+		AgentCommunicationUtils.handleRequestError(api.testStarted(uniformPath),
+				"Failed to start coverage recording for test case " + uniformPath +
+						". Coverage for that test case is lost.");
+	}
+
+	private void startTestRun() throws Exception {
+		boolean includeNonImpacted = parseAndRemoveBooleanSwitch("includeNonImpacted");
```

Comment:
Aren't command line options typically in kebab-case?

Reply:
true :-)